### DISCUSSION TITLE
v1.13 Release Candidate 2

### DISCUSSION THREAD

**Discussion by dorkster** [flareteam/flare-game#887] [Announcements]
(Note: This is list only includes changes since [1.13rc1](https://github.com/flareteam/flare-game/releases/tag/v1.13rc1).)

## Engine features

* Added 'background' property for all menus. Menus will fall back to the existing hard-coded filenames if this property is omitted.
* Added 'close', 'label_title', and 'action_list' properties to MenuConfirm configuration.
* Added 'render_layers' section to StatBlock definitions. It has the same syntax and functionality as engine/hero_layers.txt, so that file is now considered deprecated.
* Added 'animation_slots' section to StatBlock definitions. This allows for multiple animation layers on non-player entities that don't have an inventory like the player.
* Added a 'direction' property to map NPCs, indentical to the property of the same name for map enemies.
* Added 'vendor_ratio_buy', 'vendor_ratio_sell', and 'vendor_ratio_sell_old' properties to engine/loot.txt. These deprecate 'vendor_ratio' and 'vendor_ratio_buyback'.
* Added 'vendor_ratio_buy', 'vendor_ratio_sell', and 'vendor_ratio_sell_old' properties to NPCs. These can be used to override the global settings in engine/loot.txt.

## Engine fixes

* Fixed memory leak when loading NPCs.
* Entity animations will now fall back to the 'die' animation when 'critdie' isn't defined.
* Fixed several bugs with the "equipment set" feature.
* Fixed several bugs with the "fog-of-war" feature.
* Disable operating system's high-dpi scaling since Flare does its own scaling.

## Game updates

* The minimap is now gradually revealed as the player explores.
* A secondary equipment loadout has been added to the inventory.
* Uniform size and background image for small dialog windows (e.g. game over menu, number picker, etc).

## Translation updates

* Basque (eu) update (aitzkora)
* Norwegian Bokmal (nb) update (Elias Nykrem)

## Other

* A 64-bit build is now shipped as the default on Windows. The 32-bit build is still included, but is optional.

<hr /><em>This discussion was created from the release <a href='https://github.com/flareteam/flare-game/releases/tag/v1.13rc2'>v1.13 Release Candidate 2</a>.</em>

**Comment by WithinAmnesia** [comment]
This is excellent!

**Comment by WithinAmnesia** [comment]
![image](https://user-images.githubusercontent.com/86871191/150619478-b9001364-11a6-42d6-aedd-b515940fcc66.png)
I still have to get the DPI override working. I tested this with a fresh 1.13.R.C.2 64-bit Windows 10 install and the DPI was not overridden.

**Reply by dorkster** [reply]
You're right. I just tested and realized I still had the Windows compatibility override checked for flare.exe. The SDL hint seems to do nothing :/

**Reply by WithinAmnesia** [reply]
Are we at the end of our rope is there a possible solution to still do this some how? How do the other Windows 64-bit executable games with image / game screen scaling work? I think Baldur's Gate Enhanced Edition and I *_think_* Diablo 2 Resurrected can scale within the game? There has to be a way yet currently I am just not with an innate understanding of the inner-workings required to be of an enlighten perspective as required to directly alleviate this issue :-(.

**Reply by dorkster** [reply]
I looked at SDL's source code and it appears the DPI stuff only affects the following:

- Wayland (Linux)
- Cocoa (MacOS)
- UIkit (iOS)
- Emscripten (browser)

So that's a dead end. I then looked into setting it on Windows and found [this](https://docs.microsoft.com/en-us/windows/win32/hidpi/setting-the-default-dpi-awareness-for-a-process#setting-default-awareness-programmatically). Unfortunately, the MinGW enviroment I use to build only defines the old `SetProcessDPIAware()`, which is only capable of turning **on** Windows' scaling.

So for the foreseeable future, you'll have to change the compatibility setting yourself.

**Reply by WithinAmnesia** [reply]
You could throw some prototypes at my computer to test on Windows 10 if you think you might have an idea or two? I got this PC just to make games with (hopefully anyway XD) and if I can use it to run some crash tests in the pursue to make the engine better than I would more than happy to do so.

My brain is not quite getting what the scope of this "MinGW enviroment I use to build" means. How does this enable / restrict yourself / someone and is there a place / description as to learn what that means / the areas of interest / how that works? I am not good at coding currently although I have a general idea of work flows and perhaps the hardware side / the general idea of the interplay / inter-connectivity of ~'computer science' yet I am kind of lacking the connection from hardware / bits and early computer science and manufacturing to like modern OS coding (did you know a bit started as a physical, well bit XD https://www.youtube.com/watch?v=dI-JW2UIAG0 ?)

I kind of keep ~'swirling around' coding learning a bunch of pieces around it but I need to eventually just dive in and go for broke I think XD. I like math in general so should be able to get better at coding although it is slow going for myself at https://www.codecademy.com/learn/learn-c-sharp . Anyway what is the ~'world of understanding' around "MinGW enviroment I use to build"? I hope that don't ask too much at once lol XD.

**Reply by dorkster** [reply]
I was being a bit dumb. I got `SetProcessDpiAwarenessContext()` to work, but I had to use `#define _WIN32_WINNT 0x0A00` (Windows 10). This will need some testing (so no 1.13) since I want to make sure Flare still works on *at least* Windows 7 (which is what `_WIN32_WINNT` was set to).

![Screenshot_20220122_213043](https://user-images.githubusercontent.com/913982/150662628-20f86542-89aa-437c-85ab-4fc64d288489.png)

**Reply by WithinAmnesia** [reply]
Okay so this DPI issue can be ~'nibbled on' / ~'put on the back burner' and over time can be tinkered with to try and hopefully find a DPI solution for Windows 64-bit executable some time later after 1.13 is stable and or finalized (within reason)?

**Reply by dorkster** [reply]
Yes.

**Comment by WithinAmnesia** [comment]
![image](https://user-images.githubusercontent.com/86871191/150619575-2c172e2c-8fb4-4f61-af25-043aaaa1a22a.png)
Although once the DPI is working the Flare x 4 avatar loads in now and I hope with 16 Exabytes of accessible R.A.M. with the 64-bit Windows executable that the 7,000+ Flare x 4 entity stress test can work (hopefully). Either way this is really cool.

**Comment by Danimal696** [comment]
Pretty nice! I see a suggestion of mine got in, I specially dig the fow minimap and alternate equip sets.
